src/main/kotlin/uk/gov/justice/digital/hmpps/authorizationserver/adapter/AuthService.kt

```diff
@@ -18,19 +17,27 @@ class AuthService(
     private val log = LoggerFactory.getLogger(this::class.java)
   }
 
-  fun getServiceRoles(clientId: String): List<String> {
+  fun getServiceRoles(clientId: String): ServiceDetails {
     return webClient.get().uri("/api/services/roles/{clientId}", clientId)
       .retrieve()
-      .bodyToMono(object : ParameterizedTypeReference<List<String>>() {})
-      .defaultIfEmpty(emptyList())
+      .bodyToMono(ServiceDetails::class.java)
       .onErrorResume(WebClientResponseException::class.java) {
         log.error("Error response on attempt to retrieve service roles", it)
-        Mono.just(emptyList())
+        Mono.empty()
       }
       .onErrorResume(Exception::class.java) {
         log.error("Failed to retrieve service roles", it)
-        Mono.just(emptyList())
+        Mono.empty()
       }
       .block()!!
   }
 }
+
+data class ServiceDetails(
```

Comment:
If there is no naming clash would prefer to just call this `Service` rather than `ServiceDetails`